src/IECoreDelight/Renderer.cpp

```diff
@@ -421,7 +598,14 @@ class DelightShader : public IECore::RefCounted
 					std::string shaderFileName = g_shaderSearchPathCache.get( shader->getName() );
 					parameterList.add( "shaderfilename", shaderFileName );
 
-					for( const auto &parameter : shader->parameters() )
+					ConstCompoundDataPtr expandedParameters = renameParameters(
+						shaderFileName,
+						IECoreScene::ShaderNetworkAlgo::expandSplineParameters(
+							shader->parametersData()
+						)
+					);
```

Comment:
Since we're planning on adding various other network conversions, including trickier inter-shader ones, I suggest keeping any conversion out of this inner loop. Instead, we could do them directly on the ShaderNetwork in a new ShaderNetworkAlgo.cpp file. This will also give us somewhere to hang the USD conversions we want to do in future.

Reply:
Pulled out into a new `ShaderNetworkAlgo.cpp` in https://github.com/GafferHQ/gaffer/pull/5432/commits/0cec0a7dd3f01c812276ce3ac903110f91966b8b